engine/components-api/src/com/cloud/agent/AgentManager.java

```diff
@@ -143,7 +143,7 @@
 
     public void pullAgentOutMaintenance(long hostId);
 
-    boolean reconnect(long hostId);
+    void reconnect(long hostId) throws AgentUnavailableException;
```

Comment:
this change is breaking things. We need to look at it. @rafaelweingartner CAManagerImpl.java needs the boolean. should we take this to 4.12? I'm not taking the time now.

Reply:
Hmm.. Let me try to look into it today...

Reply:
Is the problem during compilation or execution?

Reply:
Never mind. Found the problem. I will work on it.

Reply:
I fixed the problem and opened a PR to send this whole contribution. This makes things easier for me.
PR: #2387